### ISSUE TITLE
Can't build example project for iOS simulator on new M1 Pro 

### ISSUE THREAD

**Issue by greenafrican** [rnmapbox/maps#1620]
**Describe the bug**  
I followed steps in the example and when I run `yarn ios` I get a very long build error that ends with:
```
** BUILD FAILED **


The following build commands failed:
	CompileC ~/Library/Developer/Xcode/DerivedData/RNMapboxGLExample-epeeljzwmhfbaletrdhogjzcxxen/Build/Intermediates.noindex/RNMapboxGLExample.build/Debug-iphonesimulator/RNMapboxGLExample.build/Objects-normal/x86_64/RNMapboxGLExample_vers.o ~/Library/Developer/Xcode/DerivedData/RNMapboxGLExample-epeeljzwmhfbaletrdhogjzcxxen/Build/Intermediates.noindex/RNMapboxGLExample.build/Debug-iphonesimulator/RNMapboxGLExample.build/DerivedSources/RNMapboxGLExample_vers.c normal x86_64 c com.apple.compilers.llvm.clang.1_0.compiler (in target 'RNMapboxGLExample' from project 'RNMapboxGLExample')
```

**To Reproduce**  
Exact steps from the example.
- git clone repo
- cd maps/example
- vi accesstoken (and pasted in and saved)
- yarn install
- yarn start
- yarn ios (new tab)

**Expected behavior**  
I'd expect the example to work

**Versions (please complete the following information):** 
As per current [example](https://github.com/react-native-mapbox-gl/maps/commit/4060e7848c363bba46398dc1bcdebc3f5838bb87)
 - Platform: iOS
 - Platform OS: iOS 11
 - Emulator/ Simulator: yes
 - Dev OS: macOS 12.0.1 (21A559)
 - react-native-mapbox-gl Version: off `master` branch (8.4.0)
 - Mapbox GL version: 5.9.0
 - React Native Version: 0.66.0

**Additional context**  
 - Yarn Version: 1.22.17
 - Node Version: 17.0.1
 - Cocoapods: 1.11.2
 - Xcode: 13.1 (13A1030d) 

This doesn't seem to be an issue with React Native as I've successfully managed to setup and build the RN example `AwesomeProject` using `npx react-native init AwesomeProject` from https://reactnative.dev/docs/environment-setup#creating-a-new-application 

Furthermore, adding mapbox to `AwesomeProject` above gives the following error during build:
```
ld: building for iOS Simulator, but linking in dylib built for iOS, file '.../AwesomeTSProject/ios/Pods/Mapbox-iOS-SDK/dynamic/Mapbox.framework/Mapbox' for architecture arm64
clang: error: linker command failed with exit code 1 (use -v to see invocation)

note: Removed stale file '~/Library/Developer/Xcode/DerivedData/AwesomeTSProject-dxukyknbwchtjkbkuuvlkpnfydrg/Build/Intermediates.noindex/Pods.build/Debug-iphonesimulator/YogaKit.build/Script-AD4DAF6EE474F7248DF8DBA6653DA7FC.sh'


** BUILD FAILED **


The following build commands failed:
        Ld ~/Library/Developer/Xcode/DerivedData/AwesomeTSProject-dxukyknbwchtjkbkuuvlkpnfydrg/Build/Products/Debug-iphonesimulator/AwesomeTSProject.app/AwesomeTSProject normal (in target 'AwesomeTSProject' from project 'AwesomeTSProject')
(1 failure)
```

UPDATE 1: when I try and run through Xcode I get the following error:
```
fatal error: module map file '~/Library/Developer/Xcode/DerivedData/RNMapboxGLExample-epeeljzwmhfbaletrdhogjzcxxen/Build/Products/Debug-iphonesimulator/YogaKit/YogaKit.modulemap' not found
```

UPDATE 2: (workaround) it all works if I use Rosetta for Terminal and Xcode. Not ideal at all.

**Comment by naftalibeder** [comment]
@greenafrican See [https://github.com/react-native-mapbox-gl/maps/discussions/1629] here - feel free to chime in.

**Comment by wiulma** [comment]
hi! same issue here. is there any news about merge m1 fix? I can confirm that in rosetta environment the example app works fine. I wasn't able to build suggested fork on m1, but in any case the fork is a little bit back compared to main repo.
thanks

**Comment by mfazekas** [comment]
Should be fixed by: #1703
